src/Umbraco.Core/Routing/DomainUtilities.cs

```diff
@@ -313,14 +315,10 @@ private static DomainAndUri GetByCulture(IReadOnlyCollection<DomainAndUri> domai
         /// <param name="domains">The domains.</param>
         /// <param name="uri">The uri, or null.</param>
         /// <returns>The domains and their normalized uris, that match the specified uri.</returns>
-        internal static IEnumerable<DomainAndUri> SelectDomains(IEnumerable<Domain> domains, Uri uri)
-        {
+        [return: NotNullIfNotNull(nameof(domains))]
+        internal static IEnumerable<DomainAndUri>? SelectDomains(IEnumerable<Domain>? domains, Uri uri)
```

Comment:
And we should be able to make this private, as it's not used by any of the tests anymore (it was when I did the v8 PR):
```suggestion
        private static IEnumerable<DomainAndUri>? SelectDomains(IEnumerable<Domain>? domains, Uri uri)
```